dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/relationship/hibernate/HibernateRelationshipStore.java

```diff
@@ -202,6 +203,16 @@ public List<Relationship> getByRelationshipType( RelationshipType relationshipTy
 
     }
 
+    @Override
+    public boolean existsIncludingDeleted( String uid )
+    {
+        Query query = getSession().createNativeQuery( "select count(*) from relationship where uid=:uid" );
```

Comment:
I would not use count if all we want is know whether there is a relationship with the UID and there is a possibility for more than one relationship with the UID (so no unique index).

Something like

`select uid from relationship where uid=:uid limit 1;`

would be enough I assume